src/Core/Managed/Shared/Extensibility/Implementation/OperationTelemetry.cs

```diff
@@ -32,6 +32,12 @@ public DateTimeOffset StartTime
         /// </summary>  
         public abstract string Id { get; set; }
 
+        /// <summary>  
+        /// Gets or sets TimeSpan representing begin time of the operation. Used internally
+        /// for calculating duration.
+        /// </summary>  
+        public TimeSpan BeginTime { get; set; }
```

Comment:
Not sure if this should be public or if this is the right place to store the time.